tplinkrouterc6u/client.py

```diff
@@ -1428,11 +1428,136 @@ def _prepare_data(self, data: str, is_login: bool) -> tuple[str, str]:
+                     format(self.__class__.__name__, e, response))
+            raise ClientException(error)
+
+    def logout(self) -> None:
+        raise ClientException('Not Implemented')
+
+    def get_firmware(self) -> Firmware:
+        data = self._request({
+            'method': 'get',
+            'device_info': {
+                'name': 'info',
+            },
+        })
+        dev_info = data['device_info']['info']
+        return Firmware(dev_info['hw_version'], dev_info['device_model'], dev_info['sw_version'])
+
+    def get_status(self) -> Status:
+        data = self._request({
+            'method': 'get',
+            'hosts_info': {
+                'table': 'host_info',
+            },
+            'network': {
+                'name': [
+                    'wan_status',
+                    'lan',
+                ],
+            },
+            'wireless': {
+                'name': [
+                    'wlan_bs',
+                    'wlan_host_2g',
+                    'wlan_wds_2g',
+                    'wlan_host_5g',
+                    'wlan_wds_5g',
+                ]
+            },
+            'guest_network': {
+                'name': [
+                    'guest_2g',
+                ]
+            }
+        })
+
+        status = Status()
+        status._wan_ipv4_addr = get_ip(data['network']['wan_status']['ipaddr'])
+        status._lan_ipv4_addr = get_ip(data['network']['lan']['ipaddr'])
+        status._lan_macaddr = get_mac(data['network']['lan']['macaddr'])
+        status.wifi_2g_enable = data['wireless']['wlan_host_2g']['enable'] == '1'
+        status.wifi_5g_enable = (data['wireless']['wlan_host_5g']['enable'] == '1' or
+                                 data['wireless']['wlan_bs']['bs_enable'] == '1')
+        status.guest_2g_enable = data['guest_network']['guest_2g']['enable'] == '1'
+
+        for item_map in data['hosts_info']['host_info']:
+            item = item_map[next(iter(item_map))]
+            dev = Device(Connection.WIRED if item['type'] == '0' else Connection.UNKNOWN, get_mac(item['mac']),
+                         get_ip(item['ip']), unquote(item['hostname']))
+            dev.up_speed = item['up_speed']
+            dev.down_speed = item['down_speed']
+            status.devices.append(dev)
+        return status
+
+    def reboot(self) -> None:
+        raise ClientException('Not Implemented')
+
+    def set_wifi(self, wifi: Connection, enable: bool) -> None:
+        raise ClientException('Not Implemented')
+
+    def _request(self, payload: dict) -> dict:
+        url = '{}/stok={}/ds'.format(self.host, self._stok)
+        response = self._session.post(url, json=payload, timeout=self.timeout, verify=self._verify_ssl)
+        return response.json()
+
+    @staticmethod
+    def _encode_password(pwd: str) -> str:
+        return TPLinkXDR3010Client._security_encode(
+            pwd,
+            'RDpbLfCPsJZ7fiv',
```

Comment:
Are you sure that these two strings are constant? Maybe there is a route to get them? Like this one https://github.com/AlexandrErohin/TP-Link-Archer-C6U/blob/main/tplinkrouterc6u/client.py#L184

Reply:
I'm not sure, because I don't think I have two XDR3010 routes.
This key is hardcoded in the JS file instead of being in the API, and I think TP Link's engineers wouldn't dynamically generate the JS file.

Reply:
If it in the JS file - agree